src/graph/optimizer/Optimizer.cpp

```diff
@@ -206,37 +215,47 @@ Status Optimizer::rewriteArgumentInputVarInternal(PlanNode *root,
 
   if (root->kind() == PlanNode::Kind::kLoop) {
     auto loop = static_cast<Loop *>(root);
-    NG_RETURN_IF_ERROR(rewriteArgumentInputVar(const_cast<PlanNode *>(loop->body())));
+    NG_RETURN_IF_ERROR(
+        rewriteArgumentInputVar(const_cast<PlanNode *>(loop->body()), visitedPlanNode));
   }
 
   if (root->kind() == PlanNode::Kind::kSelect) {
     auto sel = static_cast<Select *>(root);
-    NG_RETURN_IF_ERROR(rewriteArgumentInputVar(const_cast<PlanNode *>(sel->then())));
-    NG_RETURN_IF_ERROR(rewriteArgumentInputVar(const_cast<PlanNode *>(sel->otherwise())));
+    NG_RETURN_IF_ERROR(
+        rewriteArgumentInputVar(const_cast<PlanNode *>(sel->then()), visitedPlanNode));
+    NG_RETURN_IF_ERROR(
+        rewriteArgumentInputVar(const_cast<PlanNode *>(sel->otherwise()), visitedPlanNode));
   }
 
   return Status::OK();
 }
 
 // static
-Status Optimizer::rewriteArgumentInputVar(PlanNode *root) {
+Status Optimizer::rewriteArgumentInputVar(PlanNode *root,
+                                          std::unordered_set<const PlanNode *> &visitedPlanNode) {
```

Comment:
`visitedPlanNode` only needs to be defined inside this function, and does not need to be passed in as a parameter

Reply:
rewriteArgumentInputVar & rewriteArgumentInputVarInternal. is static function
so visitedPlanNode is defined outside the function